File: day01_2.py
def main(paren_line):
    floor = 0
    for i, c in enumerate(paren_line):
        if c == '(':
            floor += 1
        else:
            floor -=1
        if floor == -1:
            break
    return i + 1


if __name__ == '__main__':
    with open('data/input01.txt') as f:
        content = f.read().strip()
    print(main(content))

File: node/day01_2.js
'use strict';

var fs = require("fs");

const file_string = fs.readFileSync('../data/input01.txt').toString();
const string_array = file_string.split("");

let floor = 0;
let i = 0;

while (floor >= 0) {
    if (string_array[i] === '(') {
        floor += 1;
    } else {
        floor -= 1;
    }
    i += 1;
}

console.log(i);
